tests/utils/test_url.py

```diff
@@ -0,0 +1,135 @@
+from unittest.mock import MagicMock, patch
+
+import requests
+from page_analyzer.utils.url import (
+    check_url,
+    normalize,
+    parse_page_data,
+    validate,
+)
+
+
+def test_validate():
+    # given
+    url = "https://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {}
+
+
+def test_validate_invalid():
+    # given
+    url = "htttps://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {"name": "Некорректный URL"}
+
+
+def test_normalize():
+    # given
+    url = "http://www.example.com:1030/software/index.html"
+
+    # when
+    normalized_url = normalize(url)
+
+    # then
+    assert normalized_url == "http://www.example.com"
+
+
+def test_check_url():
+    # given
+    url = "https://example.com"
+
+    request_response_mock = MagicMock()
+    request_response_mock.status_code = 200
+    request_response_mock.content = (
+        "<title>some title</title>"
+        "<meta name='description' content='some description'></meta>"
+        "<h1>some header</h1>"
+    )
+
+    # when
+    with patch(
+        "page_analyzer.utils.url.requests.get", return_value=request_response_mock
+    ) as mock_requests:
+        check_result = check_url(url)
+
+        # then
+        mock_requests.assert_called_once_with(url)
+        assert check_result == {
+            "response_code": "200",
+            "h1": "some header",
+            "title": "some title",
+            "description": "some description",
+        }
```

Comment:
then part and patch asserts are usually done outside of patch context manager
```suggestion
    # then
    mock_requests.assert_called_once_with(url)
    assert check_result == {
        "response_code": "200",
        "h1": "some header",
        "title": "some title",
        "description": "some description",
    }
```